src/main/java/zmq/ZMQ.java

```diff
@@ -822,14 +822,23 @@ public Selector get()
             return selector;
         }
 
+        void close()
+        {
+            if (selector != null) {
+                try {
+                    selector.close();
+                    selector = null;
+                }
+                catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
         @Override
         public void finalize()
         {
```

Comment:
I think I'll create a separate PR building on top of this one that removes the PollSelector completely and forces the user to only create Selectors (or Pollers) via a context.

This would be a breaking change, but I think it's a step in the right direction. I do think it should be a separate PR, though, so that this PR can be strictly non-breaking changes.